File: tests/basic/object.test.ts

import ClassManager from "@base/ClassLoader"
import { createNewTestClasses } from "@base/Stubs"
import TestSystem from "@stub/tests/TestSystem"
import { expect, test } from "bun:test"

test("enum lookup and methods", () => {
	const className = "tests/basic/ObjectTest"
	const stubClasses = createNewTestClasses()

	const classManager = new ClassManager(stubClasses)
	const system = stubClasses.getStubClass("java.lang.System") as TestSystem
	const consoleOutput = system.out

	const main = classManager.get(className)
	main.executeMethod("main", classManager)

  expect(consoleOutput.printlnLines[0]).toBe("Hi ! I'm Walter White and I'm 69 years old.")
  expect(consoleOutput.printlnLines[1]).toBe("{Class A:Walter White}")
})

File: tests/basic/ObjectTest.java
package tests.basic;

public class ObjectTest {

  public static void main(String[] args) {

    B person = new B("Walter White", 69);
    System.out.println("Hi ! I'm " + person.getName() + " and I'm " +
                       person.getAge() + " years old.");
    System.out.println(person);
  }

  public static class A {

    private String name;

    public A(String name) { this.name = name; }

    /**
     * @return the name
     */
    public String getName() { return name; }

    @Override
    public String toString() {
      return "{Class A:" + this.name + "}";
    }
  }

  public static class B extends A {

    private int age;

    public B(String name, int age) {
      super(name);
      this.age = age;
    }

    /**
     * @return the age
     */
    public int getAge() { return age; }
  }
}
